Source/Core/Core/HW/WiimoteReal/WiimoteReal.cpp

```diff
@@ -406,27 +406,27 @@ static unsigned int CalculateWantedBB()
 
 void WiimoteScanner::WantWiimotes(bool do_want)
 {
-	m_want_wiimotes = do_want;
+	m_want_wiimotes.store(do_want);
 }
 
 
 void WiimoteScanner::WantBB(bool do_want)
 {
-	m_want_bb = do_want;
+	m_want_bb.store(do_want);
 }
 
 void WiimoteScanner::StartScanning()
 {
-	if (!m_run_thread)
+	if (!m_run_thread.load())
 	{
-		m_run_thread = true;
+		m_run_thread.store(true);
 		m_scan_thread = std::thread(&WiimoteScanner::ThreadFunc, this);
 	}
```

Comment:
Not sure it is necessary in this case, but this would be safer:

```c++
if (std::compare_and_exchange_weak(m_run_thread, false, true))
    m_scan_thread = std::thread(&WiimoteScanner::ThreadFunc, this);
```